src/transformers/generation_utils.py

```diff
@@ -392,7 +392,7 @@ def generate(
 
             # get encoder and store encoder outputs
             encoder = self.get_encoder()
-            encoder_outputs: tuple = encoder(input_ids, attention_mask=attention_mask)
+            encoder_outputs = encoder(input_ids, attention_mask=attention_mask, return_dict=True)
```

Comment:
Can't we guarantee that the type is `ModelOutput` now?

Reply:
line 423 makes that assumption.

Reply:
Good point! Will update this